### DISCUSSION TITLE
Zep Memory - Conversational Retrieval Agent vs Conversational Agent

### DISCUSSION THREAD

**Discussion by fmoraes87** [FlowiseAI/Flowise#1386] [Q&A]
Hey, I was doing a test and I realized when I'm using Conversational Agent, it respects the chatId or sessionId related to it, but, when I'm using Conversational Retrieval Agent it is using the executor memory (see code below) and it doesn't respect the chatId or sessionId. 

Does anybody know why it using the memory from the executor instead of getting it from nodeData?

I've changed the Conversational Retrieval Agent to test and get memory from node data instead of getting it from the executor and apparently, it works well.

Ps: Maybe I'm using it wrongly too. 

From Conversational Agent:
```

 async run(nodeData: INodeData, input: string, options: ICommonObject): Promise<string> {
        const executor = nodeData.instance as AgentExecutor
        const memory = **nodeData.inputs?.memory as BaseChatMemory**

        if (options && options.chatHistory) {
            const chatHistoryClassName = memory.chatHistory.constructor.name
            // Only replace when its In-Memory
            if (chatHistoryClassName && chatHistoryClassName === 'ChatMessageHistory') {
                memory.chatHistory = mapChatHistory(options)
                **executor.memory = memory**
            }
        }
```

From  Conversational Retrieval Agent 
```

 async run(nodeData: INodeData, input: string, options: ICommonObject): Promise<string> {
...
       
if (executor.memory) {
            // ;(executor.memory as any).memoryKey = 'chat_history'
            ;(executor.memory as any).outputKey = 'output'
            ;(executor.memory as any).returnMessages = true

            const chatHistoryClassName = **(executor.memory as any).chatHistory.constructor.name**
            // Only replace when its In-Memory
            if (chatHistoryClassName && chatHistoryClassName === 'ChatMessageHistory') {
                ;(executor.memory as any).chatHistory = mapChatHistory(options)
            }
        }
```

Thanks!

**Comment by HenryHengZJ** [comment]
@fmoraes87 noticed this issue as well. we've completely changed the approach in this [PR](https://github.com/FlowiseAI/Flowise/pull/1497) the issue should be resolved once PR is merged
